Sources/CMakeLists.txt

```diff
@@ -110,6 +113,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     add_definitions(-Wno-switch)
     add_definitions(-Wno-tautological-undefined-compare)
     add_definitions(-Wno-c++11-compat-deprecated-writable-strings)
+    add_definitions(-Wno-logical-op-parentheses)
```

Comment:
Agreed. Some of them are almost certainly wrong--albeit in harmless ways, probably--but a lot of them are difficult to decide the original programmer's intent...or the game relies on them being wrong, etc.

At the minimum, we should just put in the parentheses that make Clang happy eventually, but let's do that in another PR some other time. For now, I'll settle for a MESSAGE(WARNING "reenable this warning some day") line in CMakeLists.txt ... so we don't forget to revisit, but the warning spam during the build goes away for now.